sql/src/main/java/io/crate/operation/merge/PagingIterator.java

```diff
@@ -45,4 +52,30 @@
      * @return an iterable that will iterate through the already emitted items and emit them again in the same order as before
      */
     Iterable<TRow> repeat();
+
+    /**
+     * Returns the suitable {@link PagingIterator} according to the use case.
+     * If requiresRepeat is true then the PagingIterator is wrapped with {@link RamAccountingPageIterator}
+     * which calculates the memory usage and applies CircuitBreaker logic.
+     */
+    static PagingIterator<Integer, Row> create(int numUpstreams,
+                                               boolean requiresRepeat,
+                                               @Nullable PositionalOrderBy orderBy,
+                                               Supplier<RowAccounting> rowAccountingSupplier) {
+        PagingIterator<Integer, Row> delegatePagingIterator;
```

Comment:
minor: Would just call it `pagingIterator` - the `delegate` prefix doesn't add much readability here - if at all